model/map-file/src/main/java/org/keycloak/models/map/storage/file/common/BlockContext.java

```diff
@@ -0,0 +1,231 @@
+/*
+ * Copyright 2022 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.keycloak.models.map.storage.file.common;
+
+import org.keycloak.models.map.common.UndefinedValuesUtils;
+import org.keycloak.models.map.storage.file.yaml.YamlParser;
+import java.util.Collection;
+import java.util.LinkedHashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+import static org.keycloak.models.map.common.CastUtils.cast;
+
+/**
+ * A class implementing a {@code YamlContext} interface represents a transformer
+ * from a primitive value / sequence / mapping representation as declared in YAML
+ * format into a Java object of type {@code V}, with ability to produce
+ * the {@link #getResult() resulting instance} of parsing.
+ *
+ * <p>
+ * NOTE: In the future, this transformer might also cover the other direction:
+ * conversion from Java object into YAML primitive value / sequence / mapping representation.
```

Comment:
Isn't this note outdated? It seems to me that with `void writeValue(V value, WritingMechanism mech);` method it covers the other direction already.